### PR TITLE
feat: theming tweaks

### PR DESCRIPTION
**Pull request by dsmmcken**
Miscellaneous light and dark theme tweaks after using it for awhile. Mostly things that were either improperly styled or issues for enterprise.

- light disabled color for secondary buttons
- adjust subdued-content-bg defninition
- adjusted dark surface color
- tweaked light theme to be grayer
- added subdued-content-bg variable
- fix button alignment
- style tweaks
- minor color fixes
- style editor in conditional formating style fixes
- fix console button disabled object styling
- Numerous color adjustments
- fix malformed hsl variable
- adjust time slider styling
- fix HCM styling
- fix styling of custom time select
- item-list styling
- add fallback colors for a few more things, so that error message before api works
- Buttons can use gap instead of margin on icons now that they are flex. Much easier.
- whitespace change
- expose is modified styling to styleguide timeslider for checking
- styleguide icon button styling
- add some spectrum stuff to styleguide for easier comparison
- purge primary-light
- schotcut settings rror message styling
- settings menu user image and logo button adjust
- widget list hover color
- add legend color

### CODE REVIEW

File: packages/components/scss/bootstrap_overrides.scss [hunk 1]
```diff
@@ -180,9 +181,10 @@ $input-btn-focus-color: hsla(var(--dh-color-accent-hsl), 0.35);
 $input-btn-focus-box-shadow: 0 0 0 $input-btn-focus-width $input-btn-focus-color;
 $input-btn-line-height: 1.3;
 // Bootstrap uses a calc expression to determine the input height (calc(line-height + 2*padding-y + border)).
-// The padding-y value has to be in rem to match units. Using 0.4214rem with 1.3
-// line height gets us to 31.99px vs Spectrum's 32px.
-$input-btn-padding-y: 0.4214rem;
+// The padding-y value has to be in rem to match units as it is a SASS calc.
+// Using 0.42145rem with 1.3 line height gets us to spectrum's 32px.
+/* stylelint-disable-next-line number-max-precision */
+$input-btn-padding-y: 0.42145rem;
```
Comment (bmingles) [inline packages/components/scss/bootstrap_overrides.scss:187]: Nice tweak

File: packages/components/src/theme/theme-spectrum/theme-spectrum-overrides.css [hunk 2]
```diff
@@ -28,3 +28,40 @@ label[class^='spectrum-'] {
...
   );
 }
+
+[class^='spectrum-Table-headCell'] {
+  /* 
+  Tables are styled with the assumption that the header is the same color as the bg
+  which means they look bad against any other background color, instead just inherit the bg.
+  There is also no variable exposed for this, so we have to target the class directly.
+ */
+  --dh-table-header-cell-background-color: transparent;
```
Comment (bmingles) [inline packages/components/src/theme/theme-spectrum/theme-spectrum-overrides.css:38]: Kudos for introducing this pattern to make it possible to override

File: packages/components/src/theme/theme-spectrum/theme-spectrum-overrides.css [hunk 3]
```diff
@@ -28,3 +28,40 @@ label[class^='spectrum-'] {
...
+[class*='spectrum-Table--quiet'] {
+  /* Quiet shouldn't assume a bg color */
+  --spectrum-alias-background-color-default: transparent;
+}
+
+button[class^='spectrum-Button'] {
+  /* make the icon closer to the text */
+  --spectrum-button-primary-text-gap: 6px;
+
+  /* center the text vertically, we have different font, it requires different centering */
```
Comment (bmingles) [inline packages/components/src/theme/theme-spectrum/theme-spectrum-overrides.css:52]: May be worth clarifying that "different font" is with respect to Spectrum's font

**Review by bmingles — CHANGES_REQUESTED** [review]
Changes look good overall.
- 1 minor suggestion for a comment tweak
- Also 2 styleguide alignment things I noticed with spectrum-forms and spectrum comparison checkbox / radio examples

**Review by bmingles — DISMISSED** [review]
LGTM

**Review by mofojed — DISMISSED** [review]

File: packages/code-studio/src/styleguide/TimeSliderInputs.tsx [hunk 4]
```diff
@@ -41,10 +47,16 @@ class TimeSliderInputs extends PureComponent<
         <TimeSlider
           startTime={startTime}
           endTime={endTime}
+          isStartModified={isStartModified}
+          isEndModified={isEndModified}
           onChange={this.handleSliderChange}
         />
-        <p>StartTime: {TimeUtils.formatTime(startTime)}</p>
-        <p>EndTime: {TimeUtils.formatTime(endTime)}</p>
+        <p style={isStartModified ? { color: 'var(--dh-color-modified)' } : {}}>
```
Comment (mofojed) [inline packages/code-studio/src/styleguide/TimeSliderInputs.tsx:54]: Should we be using style tag directly here or a CSS class name and put this in the CSS?

**Review by mofojed — DISMISSED** [review]

**Review by bmingles — APPROVED** [review]

**Review by mofojed — APPROVED** [review]

**Pull request merged**